CONTRIBUTING.md

```diff
@@ -0,0 +1,99 @@
+# Contributing
+
+As a contributor, here are the guidelines we would like you to follow:
+
+- [Coding Rules](#markdown-header-coding-rules)
+- [Working with branches](#markdown-header-working-with-branches)
+- [Commit Message Guidelines](#markdown-header-commit=message-format)
+
+## Coding Rules
+
+TODO
+
+## Working with branches
+
+Remote branches must follow the following format:
+
+- fix/my-fix-name: for bugfixes/hotfixes
+- feature/my-feature-name: for features, refactoring, etc
+
+Branches name **must be in kebab-case and lower caps**.
+
+Please **keep the commit history of your branch clean**, with proper commit messages (see below).
+Use `git log` to review it, and if necessary `git rebase` to clean it.
+
+> See the [git docs](https://git-scm.com/docs/git-rebase#_interactive_mode) for more details.
+
+## Commit Message Format
+
+_This specification is inspired by [Angular commit message format](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-format)._
+
+We have very precise rules over how our Git commit messages must be formatted.
+This format leads to **easier to read commit history**.
+
+Each commit message consists of a **header** and a **body**
+
+```
+<header>
+<BLANK LINE>
+<body>
+```
+
+The `header` is mandatory and must conform to the [Commit Message Header](#markdown-header-commit-message-header) format.
+
+The `body` is mandatory for all commits except for those of type "docs".
+When the body is present it must be at least 20 characters long and must conform to the [Commit Message Body](#markdown-header-commit-message-body) format.
+
+Any line of the commit message cannot be longer than 100 characters.
+
+### Commit Message Header
+
+```
+<type>(<scope>): <short summary>
+  │        |        │
+  │        |        └─⫸ Summary in present tense. Not capitalized. No period at the end.
+  |        |
+  |        └─⫸ Commit Scope: domain-challenge|domain-resource|lib-logger|lib-common|lib-secrets
```

Comment:
the scope names for this repo needs to be 
1. common
2. nosql
3. relational
4. challenge
5. payment
6. resource